core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -138,13 +138,16 @@ private[spark] abstract class MapOutputTracker(conf: SparkConf) extends Logging
    *         and the second item is a sequence of (shuffle block id, shuffle block size) tuples
    *         describing the shuffle blocks that are stored at that block manager.
    */
-  def getMapSizesByExecutorId(shuffleId: Int, startPartition: Int, endPartition: Int)
+  def getMapSizesByExecutorId(shuffleId: Int, startPartition: Int, endPartition: Int,
+                              mapid: Int = -1)
```

Comment:
it's a good idea. the seq[Int] parameter can fetch more maps data. it can reduce the task num